PA193_mnemonic_Slytherin/mnemoniccli.py

```diff
@@ -230,14 +231,13 @@ def action_generate(config: Config) -> ExitCode:
         return ExitCode.EX_NOINPUT
     if config.format is Config.Format.TEXT_HEXADECIMAL:
         entropy = unhexlify(entropy)  # type: bytes
-    if not is_valid_entropy(entropy):
+    try:
+        entropy = Entropy(entropy)
+    except ValueError:
         msg = 'invalid entropy'
         logger.critical(msg)
         print(msg, file=sys.stderr)
         return ExitCode.EX_DATAERR
```

Comment:
You can do
~~~py
   try:
        entropy = Entropy(entropy)
    except ValueError as e:
        logger.critical(str(e))
        print(str(e), file=sys.stderr)
        return ExitCode.EX_DATAERR
~~~

Reply:
done, also extracted the error handling into decorator to reduce code repetition